File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.cpp
#include <string>
#include <vector>

using namespace std;
long long input;
int solution(int num) {
    input = num;
    for(int i = 0; i < 500; i++){
        if(input == 1)
            return i;
        else {
            if(input % 2 == 0){
                input /= 2;
            }
            else {
                input *= 3;
                input++;
            }
        }
    }
    return -1;
}

File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.java
class Solution {
    public int solution(int num) {
        long input = (long) num;
        for(long i = 0; i < 500; i++){
            if(input == 1)
                return (int)i;
            else{
                if(input % 2 == 0)
                    input /= 2;
                else{
                    input *= 3;
                    input++;
                }
            }
        }
        return -1;
    }
}